### PR TITLE
fix(shared_sub): no need for remote subscriber aliveness track

### PR DESCRIPTION
**Pull request by zmstone**
Fixes https://github.com/emqx/emqx/issues/13535

Release version: v/e5.8.1

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_auth_kerberos/test/emqx_authn_kerberos_SUITE.erl [hunk 1]
```diff
@@ -69,6 +69,26 @@ t_create(_Config) ->
     {ok, [#{provider := emqx_authn_kerberos}]} =
         emqx_authn_chains:list_authenticators(?GLOBAL).
 
+t_disable_without_kinit(_Config) ->
```
Comment (thalesmg) [inline apps/emqx_auth_kerberos/test/emqx_authn_kerberos_SUITE.erl:72]: This seems unrelated with changes herein?

File: apps/emqx_auth_kerberos/src/emqx_authn_kerberos.erl [hunk 2]
```diff
@@ -47,6 +47,8 @@ resolve_keytab(<<"DEFAULT">>) ->
 resolve_keytab(Path) ->
     emqx_schema:naive_env_interpolation(Path).
 
+update(#{enable := false}, State) ->
```
Comment (thalesmg) [inline apps/emqx_auth_kerberos/src/emqx_authn_kerberos.erl:50]: This is from the other PR?

**Review by thalesmg — APPROVED** [review]

File: apps/emqx_auth_kerberos/src/emqx_authn_kerberos.erl (hunk 2 above)
Comment (zmstone) [reply apps/emqx_auth_kerberos/src/emqx_authn_kerberos.erl:50]: i changed merge base.

File: apps/emqx_auth_kerberos/test/emqx_authn_kerberos_SUITE.erl (hunk 1 above)
Comment (zmstone) [reply apps/emqx_auth_kerberos/test/emqx_authn_kerberos_SUITE.erl:72]: this is because i changed merge base from release-58 to master.
they should be merged anyway.
